types/underscore/underscore-tests.ts

```diff
@@ -486,6 +486,28 @@ _.chain(anyValue)
     .find(i => i.findBooleanFunction())
     .value();
 
+// verify that inline partial objects can be provided without error to where and findWhere for a union type collection
+// where no types in the union share the same property names
+declare const nonIntersectinglTypeUnion: _.Dictionary<{ one: string; } | { two: number; }>;
+
+// $ExpectType { one: string; } | { two: number; } | undefined
+_.chain(nonIntersectinglTypeUnion)
+    .sample(5)
+    .where({ one: 'one' })
+    .findWhere({ two: 2 })
+    .value();
```

Comment:
This is contradictory code. I wouldn't want to have this in a test unless it is to confirm that TS will reject it. I suggest teasing the calls apart:

```suggestion
// $ExpectType ({ one: string; } | { two: number; })[]
_.chain(nonIntersectinglTypeUnion)
    .sample(5)
    .where({ one: 'one' })
    .value();

// $ExpectType { one: string; } | { two: number; } | undefined
_.chain(nonIntersectinglTypeUnion)
    .sample(5)
    .findWhere({ two: 2 })
    .value();
```

Reply:
I agree that while these might currently work from a types perspective it isn't a good example to provide in general and might start to fail in the future from changes that it might be correct to make, so I've updated them as you suggested. Thanks!